src/gordon_gcp/plugins/service/gdns_publisher.py

```diff
@@ -48,6 +50,101 @@
 DNS_CHANGES_DONE = 'done'
 
 
+class GDNSPublisherBuilder:
+    """Build and configure a :class:`GDNSPublisher` object.
+
+    Args:
+        config (dict): Google Cloud DNS API related configuration.
+        success_channel (asyncio.Queue): queue to place a successfully
+            published message to be further handled by the ``gordon``
+            core system.
+        error_channel (asyncio.Queue): queue to place a message met
+            with errors to be further handled by the ``gordon`` core
+            system.
+        kwargs (dict): Additional keyword arguments to pass to the
+            publisher.
+    """
+    def __init__(self, config, success_channel, error_channel, **kwargs):
+        self.config = config
+        self.success_channel = success_channel
+        self.error_channel = error_channel
+        self.kwargs = kwargs
+
+    def _validate_config(self):
```

Comment:
I'd like to do what I did in the enricher so it's easier to subclass (individual test methods and one caller method).